code/modules/antagonists/changeling/powers/linglink.dm

```diff
@@ -54,22 +52,23 @@
 				for(var/mob/M in GLOB.mob_list)
 					if(GLOB.all_languages["Changeling"] in M.languages)
 						to_chat(M, "<i><font color=#800080>We can sense a foreign presence in the hivemind...</font></i>")
-				target.mind.linglink = 1
+				target.mind.linglink = TRUE
 				target.add_language("Changeling")
 				target.say(":g AAAAARRRRGGGGGHHHHH!!")
 				to_chat(target, "<font color=#800040><span class='boldannounce'>You can now communicate in the changeling hivemind, say \":g message\" to communicate!</span>")
 				target.reagents.add_reagent("salbutamol", 40) // So they don't choke to death while you interrogate them
-				sleep(1800)
+				sleep(180 SECONDS)
 		SSblackbox.record_feedback("nested tally", "changeling_powers", 1, list("[name]"))
-		if(!do_mob(user, target, 20))
+		if(!do_mob(user, target, 2 SECONDS))
 			to_chat(user, "<span class='warning'>Our link with [target] has ended!</span>")
 			target.remove_language("Changeling")
-			changeling.islinking = 0
-			target.mind.linglink = 0
-			return
+			cling.is_linking = FALSE
+			target.mind.linglink = FALSE
+			return FALSE
 
 	target.remove_language("Changeling")
-	changeling.islinking = 0
-	target.mind.linglink = 0
+	cling.is_linking = FALSE
+	target.mind.linglink = FALSE
 	to_chat(user, "<span class='notice'>You cannot sustain the connection any longer, your victim fades from the hivemind</span>")
 	to_chat(target, "<span class='userdanger'>The link cannot be sustained any longer, your connection to the hivemind has faded!</span>")
+	return TRUE
```

Comment:
make this sleep into a timer?

Reply:
This entire ability is scuffed, and I just realized it calls `SSblackbox.record_feedback("nested tally", "changeling_powers", 1, list("[name]"))` ***three*** times due to the `for(var/stage in 1 to 3)`. Time to refactor.